File: services/core-auth/app/models/service.py
from __future__ import annotations

import uuid
from datetime import datetime

from sqlalchemy import DateTime, String, text
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.core.orm_base import Base


class Service(Base):
    """Catalog of delivery channels (e.g. email, sms)."""

    __tablename__ = "services"

    id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True), primary_key=True, default=uuid.uuid4
    )
    name: Mapped[str] = mapped_column(String(64), nullable=False, unique=True)
    description: Mapped[str | None] = mapped_column(String(512), nullable=True)
    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True),
        nullable=False,
        default=datetime.utcnow,
        server_default=text("now()"),
    )

    quotas: Mapped[list["Quota"]] = relationship(
        "Quota", back_populates="service_row"
    )
    usage_records: Mapped[list["Usage"]] = relationship(
        "Usage", back_populates="service"
    )

File: services/admin-panel/models/service.js
import { DataTypes } from 'sequelize'
import { sequelize } from './db.js'

export const Service = sequelize.define(
  'Service',
  {
    id: {
      type: DataTypes.UUID,
      primaryKey: true,
      defaultValue: DataTypes.UUIDV4,
    },
    name: { type: DataTypes.STRING(64), allowNull: false, unique: true },
    description: { type: DataTypes.STRING(512), allowNull: true },
    created_at: {
      type: DataTypes.DATE,
      allowNull: false,
      defaultValue: DataTypes.NOW,
    },
  },
  {
    tableName: 'services',
    timestamps: false,
    freezeTableName: true,
  }
)
